GraveRobber fixes, Infection and H2X Cleanup

Berthalamew opened the pull request.

Berthalamew committed:
Rename HeadHunter to GraveRobber

Berthalamew committed:
Rename everything related from headhunter to graverobber

Berthalamew committed:
Cleanup and fix graverobber

Berthalamew committed:
Create sound_classes_definition.hpp

Berthalamew committed:
Cleanup H2X

- forgot to rename an include in sound_classes_definition

Berthalamew committed:
Cleanup Infection

- add endl to GraveRobber, H2X, Infection

Berthalamew committed:
Fix issue with RPS based weapons

Berthalamew committed:
Update Version

nukeulater commented on xlive/H2MOD/Variants/H2X/H2X.cpp:
```diff
@@ -43,91 +42,47 @@ void H2X::ApplyMapLoadPatches(bool enable)
 	}
 
-	if (enable && !Memory::IsDedicatedServer() && h2mod->GetEngineType() == e_engine_type::_multiplayer)
+	if (enable && !Memory::IsDedicatedServer() && s_game_globals::game_is_multiplayer())
```
`h2mod->GetEngineType() == e_engine_type::_multiplayer` was right, you are breaking this, also for the code bellow you should add a size check for the sound class blocks if possible

Berthalamew replied:
https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-53f3f1a9f579c86f48f5e72e7355f36eddde2a46ee2fed9289c1410426dba388R56
Fixed

nukeulater commented on xlive/Project_Cartographer.vcxproj.user:
```diff
@@ -7,6 +7,13 @@
     <DebuggerFlavor>WindowsLocalDebugger</DebuggerFlavor>
     <LocalDebuggerCommandArguments>-nointro -windowed</LocalDebuggerCommandArguments>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='GameRewriteDebug|Win32'">
```
please don't add this to the main repository

Berthalamew replied:
https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-c60806c8d16d9111955b303c7360c44628dff63caab58e858e8c86a84cf0c878L10-L16
Fixed

nukeulater commented on xlive/Project_Cartographer.vcxproj.user:
```diff
@@ -22,6 +29,13 @@
     <LocalDebuggerDebuggerType>NativeOnly</LocalDebuggerDebuggerType>
     <LocalDebuggerCommandArguments>-nointro  -novsync</LocalDebuggerCommandArguments>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='GameRewriteRelease|Win32'">
```
same with this, don't add to the main repository

Berthalamew replied:
https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-c60806c8d16d9111955b303c7360c44628dff63caab58e858e8c86a84cf0c878L32-L38
Fixed

nukeulater commented on xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp:
```diff
@@ -0,0 +1,242 @@
+
+	if (!DATUM_IS_NONE(skullDatum))
+	{
+		typedef char* (__cdecl* get_score_data_ptr)();
```
`get_score_data` is enough, also the type definition should use `_t` at the end

Berthalamew replied:
https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-ac5bc43b1a79579fc36772ba42d52a8f5a4cd2c045d586ce2c21b4528b0e5d79R69-R70
Changed it to get_score_data_t

nukeulater commented on xlive/H2MOD/Variants/GraveRobber/GraveRobber.cpp:
```diff
@@ -0,0 +1,242 @@
+			HaloScript::ObjectDestroy(skullDatum);
+
+			if (DATUM_INDEX_TO_ABSOLUTE_INDEX(h2mod->get_player_datum_index_from_controller_index(0)) == absPlayerIdx ||
+				DATUM_INDEX_TO_ABSOLUTE_INDEX(h2mod->get_player_datum_index_from_controller_index(1)) == absPlayerIdx ||
```
we should create a helper function for this, also should be a loop

Berthalamew replied:
https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-ac5bc43b1a79579fc36772ba42d52a8f5a4cd2c045d586ce2c21b4528b0e5d79
Changed it to a loop, and reworked the function so it doesn't have a bunch of nested if statements

nukeulater commented on xlive/H2MOD/Variants/Infection/Infection.h:
```diff
@@ -22,31 +55,20 @@ class Infection : public ICustomGameVariant
-	}
-
-	Infection();
-	~Infection();
```
deconstructor should be virtual here too
otherwise good catch

Berthalamew replied:
https://github.com/pnill/cartographer/pull/407/commits/15ddad208effb18819e21dedf764dd9bbc93430a#diff-722acf67b979a805dd7e77189f0f0f4ee9e23aff90c6145b6c05ab85371695b1R65
Changed it to virtual

Berthalamew committed:
Requested changes

- remove extra build types (thanks VS for pulling them over)
- add block check for soundclass code in H2X
- change engine type check to use h2mod in H2X
- Rework PickupSkull function so it isnt a bunch of nested if statements and make the check for local players a loop
- make Infection destructor virtual

Berthalamew committed:
More cleanup

- move sound classes fix to it's own function that's applied to every map (not just h2x)
- rename a function pointer in graverobber

Berthalamew committed:
Fix Crash when loading SP maps in the MP game engine

Berthalamew committed:
Fix bug where players were unable to pickup skulls
